src/any_llm/providers/bedrock/bedrock.py

```
@@ -66,10 +66,23 @@ class BedrockProvider(AnyLLM):
+    # `_verify_and_set_api_key` with the same explicit credentials the real client is later
+    # built with, instead of a bare `boto3.Session()` that only sees ambient credentials.
+    _SESSION_CREDENTIAL_KWARGS = (
+        "aws_access_key_id",
+        "aws_secret_access_key",
+        "aws_session_token",
+        "region_name",
+        "profile_name",
+        "botocore_session",
+    )
+
     def __init__(self, api_key: str | None = None, api_base: str | None = None, **kwargs: Any) -> None:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)